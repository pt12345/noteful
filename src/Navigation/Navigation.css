.folder_link.active {
    background-color: gray;
}