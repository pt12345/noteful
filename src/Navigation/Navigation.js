import React, { Component } from 'react'
import { NavLink, Link } from 'react-router-dom'
import './Navigation.css'
import ApiContext from '../ApiContext';

class Navigation extends Component {

  static contextType = ApiContext

  render() {
    const { folders=[] } = this.context

    return(
      <div>
          <ul>
              {folders.map(folder => 
                  <li key={folder.id}>
                      <NavLink
                        className='folder_link'
                        to={`/folder/${folder.id}`}>
                            {folder.name}
                        </NavLink>
                  </li>
              )

              }
          </ul>
          <div className='new_folder'>
              <span>New Folder</span>
          </div>
      </div>
    );
  }
}

export default Navigation;