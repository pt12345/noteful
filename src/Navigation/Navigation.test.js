import React from 'react';
import ReactDOM from 'react-dom';
import Navigation from './Navigation';
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

describe('Navigation', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<BrowserRouter><Navigation /></BrowserRouter>, div);

    ReactDOM.unmountComponentAtNode(div);
  });

  it('Renders UI as expected', () => {
    const nav = renderer.create(<BrowserRouter><Navigation /></BrowserRouter>).toJSON();
    expect(nav).toMatchSnapshot();
  })

})
