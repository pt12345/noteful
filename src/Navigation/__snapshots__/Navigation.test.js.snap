// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Renders UI as expected 1`] = `
<div>
  <ul />
  <ul
    className="new_folder"
  >
    <li
      className="new_folder_link"
    >
      <a
        aria-current={null}
        href="/folder/addFolder"
        onClick={[Function]}
      >
        Add Folder
      </a>
    </li>
  </ul>
</div>
`;
